patches/tModLoader/Terraria/ModLoader/ModPlayer.cs

```diff
@@ -284,6 +284,12 @@ public virtual void PostUpdate()
 	{
 	}
 
+	/// <summary>
+	/// Called within Player.RefreshInfoAccsFromItemType. You should set any info accessory related fields to true based on the
+	/// passed in value in this hook.
+	/// </summary>
+	public virtual void UpdateInfoAccessory(int accessoryType) { }
```

Comment:
Can you defer this to another PR? To explain a bit more about the info acc situation (after reading more closely):

- Currently, items use `ModItem.UpdateInventory`. 
- In single-player, inventory pause may be enabled, so no updates actually happen when first acquiring an accessory. 
- To mitigate this, 1.4.4 made info accs update even when the game is paused.

To achieve feature parity for modded info accessories we need:
- `ModPlayer.ResetInfoAccs`, to be called alongside `ResetEffects` and also in `Player.RefreshInfoAccs`